plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/aggregation/DataSketchState.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.aggregation;
+
+import io.trino.spi.function.AccumulatorState;
+import io.trino.spi.function.AccumulatorStateMetadata;
+import org.apache.datasketches.theta.CompactSketch;
+import org.apache.datasketches.theta.UpdateSketch;
+
+@AccumulatorStateMetadata(stateSerializerClass = DataSketchStateSerializer.class)
+public interface DataSketchState
+        extends AccumulatorState
+{
+    UpdateSketch getUpdateSketch();
+
+    void setUpdateSketch(UpdateSketch value);
+
+    CompactSketch getCompactSketch();
+
+    void setCompactSketch(CompactSketch value);
+
+    @Override
+    default long getEstimatedSize()
+    {
+        throw new UnsupportedOperationException(); // is it called at all?
```

Comment:
I see also the method used only in tests at the moment.
Would it make sense to remove it (follow-up PR)?

Reply:
This is a left over, thanks for catching this.
The method gets implemented here
https://github.com/trinodb/trino/blob/dafbc39401e34c20a625a809ecb0e46df6f445bf/core/trino-main/src/main/java/io/trino/operator/aggregation/AccumulatorCompiler.java#L319-L333